### PR TITLE
Multiple src queries

### PR DESCRIPTION
**Pull request by samiyaakhtar**
Issue #131

### CODE REVIEW

File: client/api_handler.py [hunk 1]
```diff
@@ -24,55 +25,67 @@
...
-        response_code = StatusCode.BAD_REQUEST
-        sanitized_url = url
+    json_output = collections.OrderedDict()
+    json_output['length'] = len(urls)
+    json_output['responses'] = []
+
+    for url in urls:
+        url = url.strip()
+        logger = current_app.logger
+        logger.setLevel(logging.DEBUG)
```
Comment (AliceYuan) [inline client/api_handler.py]: mind removing this line? I think I added it by accident

File: client/query_utils.py [hunk 2]
```diff
@@ -4,7 +4,8 @@
 
 class QueryParams:
 
-    def __init__(self):
+    def __init__(self, urls):
         self.response_type = ResponseType.JSON
```
Comment (MeenuV) [inline client/query_utils.py]: Why did you change the constructor?

File: client/api.py [hunk 3]
```diff
@@ -25,14 +25,14 @@ def health_check():
...
+    urls = request.args.getlist('src')
+    query_params = QueryParams(urls)
     local_request = request
 
-    url = request.args.get('src').strip()
     response_type = local_request.args.get('format')
     desc_length = local_request.args.get('desc_cap')
 
-    query_params.query_url = url
+    query_params.query_url = urls
```
Comment (MeenuV) [inline client/api.py]: You seem to be constructing the object already with the list of urls

File: client/api.py [hunk 4]
```diff
@@ -25,14 +25,14 @@ def health_check():
 
 @app.route('/website', methods=['GET'])
 def get_metadata():
-    query_params = QueryParams()
+    urls = request.args.getlist('src')
```
Comment (AliceYuan) [inline client/api.py]: just wondering does this actually work if there isn't a list of urls?
I think this would return a list even if there is only 1 url, which is not what we want.

File: client/api_handler.py [hunk 5]
```diff
@@ -24,55 +25,67 @@
...
-    content_type = None
-    response_code = None
-
-    head = url_utils.get_requests_header(sanitized_url)
-    if head is None:
-        metadata = create_metadata_object(url, StatusCode.BAD_REQUEST, None, None)
-        response_code = StatusCode.BAD_REQUEST
-        sanitized_url = url
+    json_output = collections.OrderedDict()
+    json_output['length'] = len(urls)
```
Comment (AliceYuan) [inline client/api_handler.py]: I think 'count' would make more sense in this, or `response_count`

**Comment by samiyaakhtar** [comment]
@AliceYuan it doesnt return a list when there is only 1, i can show you the outputs i get for 1 and 2 requests if you want.

File: client/api.py (hunk 3 above)
Comment (samiyaakhtar) [inline client/api.py]: I don't get, what do you mean one variable? I only changed it from url to urls, coz it can be multiple. @MeenuV

File: client/api.py (hunk 4 above)
Comment (samiyaakhtar) [inline client/api.py]: It works even if there is just one

File: client/api_handler.py [hunk 6]
```diff
@@ -24,55 +25,67 @@
...
+        json_data = get_json_metadata(metadata.data_map)
+        logger.info(json_data)
+        if config.CACHE_DATA and site_response.status_code == StatusCode.OK:
+            cache_map = metadata.get_cache_prop_map()
+            cache_data = get_json_metadata(cache_map)
+            logger.debug("Caching json data to redis db")
+            cache_utils.cache_json_data(sanitized_url, cache_data)
+        json_output['responses'].append(json.loads(json_data))
+
+    if json_output['length'] > 1:
```
Comment (AliceYuan) [inline client/api_handler.py]: I think this is a little messy, can we extract out the getting multiple urls & getting the metadata for one url to a different function?
eg:
```def get_url_metadata(query_params):
...

def get_urls_metadata(query_params):
< logic for getting many responses and making into a json call get_url_metadata>

```
In this case, we can easily make `get_url_metadata` a concurrent function.
Also, in order to make it possible to be concurrent you should make query_params an array of QueryParam objects rather than have urls in query_param.

File: client/query_utils.py (hunk 2 above)
Comment (samiyaakhtar) [inline client/query_utils.py]: I'll undo that sorry @MeenuV

File: client/api_handler.py (hunk 6 above)
Comment (MeenuV) [inline client/api_handler.py]: +1

File: client/query_utils.py (hunk 2 above)
Comment (MeenuV) [inline client/query_utils.py]: No worries, was just wondering if there was a valid reason for this change. If so, we can keep it

**Comment by samiyaakhtar** [comment]
I'll make commit after cleaning up the way you asked, Alice.

**Comment by samiyaakhtar** [comment]
Done cleaning up @AliceYuan @MeenuV

**Comment by AliceYuan** [comment]
it looks fine to me, can you add some documentation in the readme on how to maple multiple url requests? and what the structure of the response would look like

**Comment by samiyaakhtar** [comment]
Done adding documentation, and the tiny changes @AliceYuan @MeenuV

**Comment by AliceYuan** [comment]
looks good! r+
